crate/memories/src/batching_layer/buffer.rs

```diff
@@ -0,0 +1,95 @@
+use std::{mem, sync::Mutex};
+
+use crate::{
+    BatchingMemoryADT,
+    batching_layer::operation::{Operation, PendingOperations},
+};
+
+struct Buffer<M: BatchingMemoryADT> {
+    capacity: usize, // the size at which the buffer should be flushed
```

Comment:
I think the term "capacity" is clear enough. What is lacking, however, is a module-level doc explaining the inner-workings of the memory batcher.

```suggestion
    capacity: usize,
```

Reply:
done